strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactController.java

```diff
@@ -394,14 +401,25 @@ public ResponseEntity publish(@RepositoryMapping Repository repository,
         return ResponseEntity.ok("");
     }
 
+    @PreAuthorize("hasAuthority('UI_LOGIN')")
+    @PutMapping(path = "{storageId}/{repositoryId}/-/user/org.couchdb.user:{username}",
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity addUser(Authentication authentication)
+    {
```

Comment:
we also better check here that `{username}` is the same that one we have in `authentication` object

Reply:
@LeifBrooks can you please add following, like in `LoginController`:
```
        if (authentication == null || !authentication.isAuthenticated())
        {
            throw new InsufficientAuthenticationException("unauthorized");
        }
        if (!(authentication instanceof UsernamePasswordAuthenticationToken))
        {
            return toResponseEntityError("Unsupported authentication class " + authentication.getClass().getName());
        }

        Object principal = authentication.getPrincipal();
        if (!(principal instanceof SpringSecurityUser)) {
            return toResponseEntityError("Unsupported authentication principal " + Optional.ofNullable(principal).orElse(null));
        }
```

and also check that if `{username}` not the same as `((SpringSecurityUser) principal).getUsername()` then `throw new BadCredentialsException(...)`.